### PR TITLE
:window: :art: Connector Field UX Improvements

### PR DESCRIPTION
**Pull request by lmossman**
## What
Relates to https://github.com/airbytehq/airbyte/issues/12204

This PR addresses the following parts of the above ticket:
1. Moves field descriptions and examples into InfoTooltips (open to feedback on the tooltip/examples styling here!): 
    Screen Shot 2022-09-01 at 6 49 01 PM
3. Changes invalid input behavior so that instead of just highlighting the description, it adds the actual form error text below the input and uses the existing `error` prop logic on the inputs to make their border red:
    Screen Shot 2022-09-01 at 6 09 30 PM
4. Removes the confusing `*` or lack-thereof after property labels which was meant to indicate optionality, and instead adds an `Optional` text to the right of the label for any optional fields.
    - Note 1: Some field names in the connector specs themselves contain `*` or `(Optional)` at the end - the connectors should be updated to remove these redundant suffixes now that we have logic in place to add the `Optional` text after the field. Should I programmatically try to filter these strings out of the field names in the meantime?
    - Note 2: I have explicitly set the `optional` flag on the PropertyLabel to `false` for the LabeledSwitch, ConditionSection, and ArraySection components, because it felt confusing as a user to have `Optional` on each of these when there is not really a way to not use them -- the switch is always on or off, condition always has some dropdown value selected, and even if the array is empty it is still submitted as a value. So to me, optional only really seemed to make sense on the basic text input components.
    Screen Shot 2022-09-01 at 6 11 14 PM
5. Removes the logic to use the first field example as the placeholder text for its input, as suggested by Tim since he felt that it was confusing, making users think that they _had_ to use the example value.

~**TODO**: modify the styling on Condition and Array section components to more closely match the styling in the [original design](https://www.figma.com/file/etaOkOsnXMoKwbfhCD71kw/WEB-APP-%E2%9A%A0%EF%B8%8F%E2%9A%A0%EF%B8%8F-OLD-SCREENS---DO-NOT-IMPLEMENT-%E2%9A%A0%EF%B8%8F%E2%9A%A0%EF%B8%8F?node-id=35016%3A197962), with the label being fully left-justified. This will allow us to hide the bounding line when there are no children in the condition group, since that case currently looks a bit strange now that the description has been moved into the InfoTooltip, e.g.:
Currently just waiting on a final confirmation of the design from Nico before implementing this. This could also be done in a follow-up PR if the current state in this PR is acceptable.~
The styling of grouped fields has been updated to match the [design](https://www.figma.com/file/gMNRYVJav25pdsoxAyW7Qp/00_01_WEB-APP-LIBRARIES?node-id=1107%3A2444) provided by Nico: 
Screen Shot 2022-09-12 at 2 37 55 PM

**NOTE: This PR also does not do any of the field grouping / optional fields hidden behind an expandable dropdown changes present in the design.** Field grouping will require some more changes to connector specs, and hiding optional fields can be done in a follow-up PR.

## How
There are several places that I have tried to refactor or simplify some components to reduce duplication and remove unneeded logic. 

I also tried to make sure that the changes I've made here won't affect other views that are using the same components that I've touched, such as the connection Replication view. Though, the changes I've made here should hopefully make it easy to update those places to use the new styling as well (e.g. use the `PropertyLabel` class I have introduced to get the description-in-tooltip and `Optional` flag styling)

## Recommended reading order
1. `PropertySection.tsx` - changes to the base input properties
3. `PropertyLabel.tsx` - replacement of the `airbyte-webapp/src/views/Connector/ServiceForm/components/Property/Label.tsx` component, since I was getting confused at the fact that it was named the same as the `airbyte-webapp/src/components/Label/Label.tsx` component which is also used further down the component tree.
4. `LabelInfo.tsx` / `LabelInfo.module.scss` - replacement of the `LabelMessage` component, since this information is no longer being displayed in label "message" but instead in the InfoTooltip. Used by the `PropertyLabel` component to display an InfoTooltip with the field description and examples.
5. `ControlLabels.tsx` - main change here is to add the InfoTooltip and the `Optional` pieces after the field label, which are both configured through new props so that this does not modify other views that use this component
6. `ConnectorNameControl.tsx` - modified this to use the new `PropertyLabel` to get the same info tooltip styling as other inputs
7. `Control.tsx` - just removed the example-placeholder logic, and added an `error` prop which is applied to the TextArea and Input subcomponents to utilize the existing logic they have to make input borders red when there is an input error
8. The rest of the changed files are somewhat small and hopefully the diff is enough!

## 🚨 User Impact 🚨
Just visual changes to the connector page, but with all of the same functionality kept the same.

### CODE REVIEW

**Comment by Upmitt** [comment]
@lmossman I updated Figma with a new component for the group of fields : https://www.figma.com/file/gMNRYVJav25pdsoxAyW7Qp/00_01_WEB-APP-LIBRARIES?node-id=1107%3A1880 - and here: https://www.figma.com/file/AxqFXuHo2BoTY0QY12hxm1/00_02_WEB-APP-COMPONENTS?node-id=446%3A8860 (you can change the width of this group of fields on Figma)

**Comment by edmundito** [comment]
@lmossman would it be possible to add storybooks for the areas impacted?

**Comment by krishnaglick** [comment]
Looks like there's a conflict!

File: airbyte-webapp/src/views/Connector/ServiceForm/components/Property/LabelInfo.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import { JSONSchema7Type } from "json-schema";
+import React from "react";
+
+import { TextWithHTML } from "components";
+
+import styles from "./LabelInfo.module.scss";
+
+interface IProps {
```
Comment (krishnaglick) [inline airbyte-webapp/src/views/Connector/ServiceForm/components/Property/LabelInfo.tsx:8]:
```suggestion
interface Props {
```

File: airbyte-webapp/src/views/Connector/ServiceForm/components/Property/LabelInfo.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  label: React.ReactNode;
+  examples?: JSONSchema7Type;
+  description?: string;
+}
+
+const LabelInfo: React.FC<IProps> = ({ label, examples, description }) => {
```
Comment (krishnaglick) [inline airbyte-webapp/src/views/Connector/ServiceForm/components/Property/LabelInfo.tsx:14]:
```suggestion
export const LabelInfo: React.FC<Props> = ({ label, examples, description }) => {
```

File: airbyte-webapp/src/views/Connector/ServiceForm/components/Property/LabelInfo.tsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  return (
+    <>
+      <h4 className={styles.descriptionHeader}>{label}</h4>
+      {description && <TextWithHTML className={styles.description} text={description} />}
+      {constructExamples()}
+    </>
+  );
+};
+
+export { LabelInfo };
```
Comment (krishnaglick) [inline airbyte-webapp/src/views/Connector/ServiceForm/components/Property/LabelInfo.tsx:44]:
```suggestion
```

File: airbyte-webapp/src/views/Connector/ServiceForm/components/Property/LabelInfo.tsx [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+import styles from "./LabelInfo.module.scss";
+
+interface IProps {
+  label: React.ReactNode;
+  examples?: JSONSchema7Type;
+  description?: string;
+}
+
+const LabelInfo: React.FC<IProps> = ({ label, examples, description }) => {
+  const constructExamples = () => {
```
Comment (krishnaglick) [inline airbyte-webapp/src/views/Connector/ServiceForm/components/Property/LabelInfo.tsx:15]: This should be a component. I'd argue for moving it above this one, renaming to `Examples` perhaps, and passing it the props so the use should look like:
`<Examples examples={examples} />`

File: airbyte-webapp/src/views/Connector/ServiceForm/components/Sections/ArraySection.tsx [hunk 5]
```diff
@@ -64,13 +65,12 @@ export const ArraySection: React.FC<ArraySectionProps> = ({ formField, path, dis
     };
   }, [items, formField.properties]);
 
+  const label = (
```
Comment (krishnaglick) [inline airbyte-webapp/src/views/Connector/ServiceForm/components/Sections/ArraySection.tsx:68]: Nit: I'd either move this down into `title={<Property...` or call this `groupTitle` or something.

File: airbyte-webapp/src/views/Connector/ServiceForm/components/Sections/ConditionSection.tsx [hunk 6]
```diff
@@ -73,16 +74,18 @@ export const ConditionSection: React.FC<ConditionSectionProps> = ({ formField, p
...
 
   return (
     <GroupControls
       key={`form-field-group-${formField.fieldKey}`}
-      description={formField.description}
       fullWidthTitle
       title={
         <div className={styles.sectionTitle}>
-          {label ? <GroupLabel>{label}:</GroupLabel> : null}
+          {label ? <GroupLabel>{label}</GroupLabel> : null}
```
Comment (krishnaglick) [inline airbyte-webapp/src/views/Connector/ServiceForm/components/Sections/ConditionSection.tsx:88]: Could label ever be undefined now?

File: airbyte-webapp/src/views/Connector/ServiceForm/components/Sections/ConditionSection.tsx [hunk 7]
```diff
@@ -24,7 +25,7 @@ const GroupLabel = styled(Label)`
 `;
 
 const ConditionControls = styled.div`
-  padding-top: 25px;
+  padding-top: 20px;
 `;
```
Comment (krishnaglick) [inline airbyte-webapp/src/views/Connector/ServiceForm/components/Sections/ConditionSection.tsx:29]: Since we have a styles folder, we should get rid of ConditionControls and just use a style.

File: airbyte-webapp/src/views/Connector/ServiceForm/components/Property/LabelInfo.module.scss [hunk 8]
```diff
@@ -0,0 +1,34 @@
+@use "../../../../../scss/colors";
+
+.descriptionHeader {
+  margin-top: 0;
+  margin-bottom: 5px;
+}
+
+.description {
+  & ul {
```
[the remark this answers is not in the document]
Comment (lmossman) [reply airbyte-webapp/src/views/Connector/ServiceForm/components/Property/LabelInfo.module.scss:9]: I tried `ul.description` and that didn't work, but `.description ul` did!

File: airbyte-webapp/src/views/Connector/ServiceForm/components/Property/LabelInfo.tsx (hunk 4 above)
Comment (lmossman) [reply airbyte-webapp/src/views/Connector/ServiceForm/components/Property/LabelInfo.tsx:15]: I like that suggestion, thanks!

File: airbyte-webapp/src/views/Connector/ServiceForm/components/Sections/ConditionSection.tsx (hunk 6 above)
Comment (lmossman) [reply airbyte-webapp/src/views/Connector/ServiceForm/components/Sections/ConditionSection.tsx:88]: Doesn't seem like it -- I've removed the conditional here

File: airbyte-webapp/src/components/GroupControls/GroupControls.tsx [hunk 9]
```diff
@@ -1,7 +1,8 @@
 import React from "react";
 import styled from "styled-components";
 
-import { Label, TextWithHTML } from "components";
+import { Label } from "components";
+import { TextWithHTML } from "components/TextWithHTML";
```
Comment (krishnaglick) [inline airbyte-webapp/src/components/GroupControls/GroupControls.tsx:5]: Unnecessary change?

File: airbyte-webapp/src/components/index.tsx [hunk 10]
```diff
@@ -18,6 +18,5 @@ export * from "./PageTitle";
 export * from "./SimpleTableComponents";
 export * from "./Spinner";
 export * from "./StatusIcon";
-export * from "./TextWithHTML";
```
Comment (krishnaglick) [inline airbyte-webapp/src/components/index.tsx:21]: I don't think you wanted to do this?

File: airbyte-webapp/src/views/Connector/ServiceForm/components/Property/LabelInfo.module.scss [hunk 11]
```diff
@@ -5,12 +5,10 @@
   margin-bottom: 5px;
 }
 
-.description {
-  & ul {
-    padding-left: 25px;
-    margin-bottom: 5px;
-    margin-top: 5px;
-  }
+.description ul {
```
Comment (krishnaglick) [inline airbyte-webapp/src/views/Connector/ServiceForm/components/Property/LabelInfo.module.scss:8]: Works for me!

File: airbyte-webapp/src/views/Connector/ServiceForm/components/Property/LabelInfo.tsx [hunk 12]
```diff
@@ -1,44 +1,42 @@
 import { JSONSchema7Type } from "json-schema";
 import React from "react";
 
-import { TextWithHTML } from "components";
+import { TextWithHTML } from "components/TextWithHTML";
```
Comment (krishnaglick) [inline airbyte-webapp/src/views/Connector/ServiceForm/components/Property/LabelInfo.tsx:4]: If you fix the barrel this line should be reverted.

File: airbyte-webapp/src/views/Connector/ServiceForm/components/Property/LabelInfo.tsx [hunk 13]
```diff
@@ -1,44 +1,42 @@
...
+  );
+};
 
+export const LabelInfo: React.FC<LabelInfoProps> = ({ label, examples, description }) => {
   return (
     <>
       <h4 className={styles.descriptionHeader}>{label}</h4>
       {description && <TextWithHTML className={styles.description} text={description} />}
-      {constructExamples()}
+      <Examples examples={examples} />
```
Comment (krishnaglick) [inline airbyte-webapp/src/views/Connector/ServiceForm/components/Property/LabelInfo.tsx:39]: 👍

File: airbyte-webapp/src/components/GroupControls/GroupControls.tsx (hunk 9 above)
Comment (krishnaglick) [reply airbyte-webapp/src/components/GroupControls/GroupControls.tsx:5]: If you fix the barrel this can be reverted.

File: airbyte-webapp/src/components/index.tsx (hunk 10 above)
Comment (lmossman) [reply airbyte-webapp/src/components/index.tsx:21]: I thought that using a named export meant we didn't want to have this in the index anymore, but I think I misunderstood

Comment (krishnaglick) [reply airbyte-webapp/src/components/index.tsx:21]: I'm happy to do a little 🤿 dive into exports/imports. Or could talk about it @ tech time 🤔

**Comment by lmossman** [comment]
I've added storybooks for the main components that I touched (GroupControls and PropertyLabel). The positioning of the group controls border is slightly different in storybook than in the Airbyte UI for some reason - I will look at this with Tim tomorrow

**Comment by lmossman** [comment]
But I believe I have addressed all of the other outstanding things on this PR, so it should be ready for a final look @krishnaglick @edmundito @timroes @Upmitt

**Comment by krishnaglick** [comment]
Screen Shot 2022-09-13 at 9 47 23 AM
The spacing seems off here. Also happens in the app.

**Comment by lmossman** [comment]
> The spacing seems off here. Also happens in the app.

@krishnaglick yeah that is what I was seeing in storybook, as I called out [here](https://github.com/airbytehq/airbyte/pull/16152#issuecomment-1244686762). But I wasn't seeing this spacing issue happen in the Airbyte UI -- it sounds like you were able to reproduce it there? If so, which connector did you use to reproduce it, and could you share a screenshot of that as well?

**Comment by krishnaglick** [comment]
Screen Shot 2022-09-14 at 10 07 28 AM
Could use just a little more spacing perhaps?

**Comment by lmossman** [comment]
> Could use just a little more spacing perhaps?

@krishnaglick this PR is only focusing on the connector form, not the connection form. Did you maybe post the wrong screenshot?

**Comment by krishnaglick** [comment]
> > Could use just a little more spacing perhaps?
> 
> @krishnaglick this PR is only focusing on the connector form, not the connection form. Did you maybe post the wrong screenshot?

Ah! That makes a lot of sense. My bad! I'll retest the correct page.

**Comment by krishnaglick** [comment]
Some conflicts here, probably easy fixes from the React 18 upgrade prep PR.

**Comment by lmossman** [comment]
> Some conflicts here, probably easy fixes from the React 18 upgrade prep PR.

Yep you were right - just resolved the conflicts and fixed the other places where I needed PropsWithChildren, so this should be good now

**Comment by krishnaglick** [comment]
Screen Shot 2022-09-15 at 9 43 45 AM
Screen Shot 2022-09-15 at 9 45 27 AM

Source creation page. This is in Firefox. Looks cleaner on Chrome.

**Comment by lmossman** [comment]
@krishnaglick I believe I've fixed all of the styling issues you pointed out above with these commits: https://github.com/airbytehq/airbyte/pull/16152/files/e63e414b3fe45ebf0540d06df7c3c3dbed75d5ff..d39a2447e1b5662984a50a937ff1de615cf59eef

I still scheduled some time with you tomorrow to go over some of those changes over voice, to make sure I'm not doing anything weird. But if those changes look good to you then this may be ready to go

**Review by krishnaglick — APPROVED** [review]
LGTM locally, in storybook and in the app!

File: airbyte-webapp/src/views/Connector/ServiceForm/components/Property/PropertyLabel.stories.tsx [hunk 14]
```diff
@@ -0,0 +1,34 @@
+import { ComponentStory, ComponentMeta } from "@storybook/react";
+
+import { Card, Input } from "components";
+
+import { PropertyLabel } from "./PropertyLabel";
+
+export default {
+  title: "Composed/PropertyLabel",
```
Comment (edmundito) [inline airbyte-webapp/src/views/Connector/ServiceForm/components/Property/PropertyLabel.stories.tsx:8]: Connector/PropertyLabel may be a better place for this.

I would also move the ServiceForm story from views to Connector/ServiceForm

**Review by edmundito — COMMENTED** [review]
Just did an initial pass and tested locally to see that things looked good!

I mentioned it in some places but overall I'd like to see the css classnames be more concise and take advantage of the css modules conversion. The way the modules work is that it will isolate the classnames. For example, if you have the class `container` inside the `GroupControls` module, the classname will translate to `GroupControls__container`, so it does not need not 1:1 from the styled component.

File: airbyte-webapp/src/components/LabeledControl/ControlLabels.module.scss [hunk 15]
```diff
@@ -0,0 +1,17 @@
+@use "../../scss/colors";
+
+.controlContainer {
+  width: 100%;
+  display: inline-block;
```
[the remark this answers is not in the document]
Comment (lmossman) [reply airbyte-webapp/src/components/LabeledControl/ControlLabels.module.scss]: As far as I can tell, changing this from `inline-block` to `block` has no visible difference, so I just switched it to block

File: airbyte-webapp/src/components/GroupControls/GroupControls.tsx [hunk 16]
```diff
@@ -1,45 +1,20 @@
...
+const GroupControls: React.FC<React.PropsWithChildren<GroupControlsProps>> = ({ title, children, name }) => {
   return (
-    <FormGroup data-testid={name}>
-      <GroupTitle $fullWidthTitle={fullWidthTitle}>{title}</GroupTitle>
-      {description && <Label message={<TextWithHTML text={description} />} />}
-      {children}
-    </FormGroup>
+    <div className={styles.groupControlSection}>
+      <div className={styles.groupTitle}>{title}</div>
+      <div className={styles.formGroup} data-testid={name}>
```
[the remark this answers is not in the document]
Comment (lmossman) [reply airbyte-webapp/src/components/GroupControls/GroupControls.tsx]: Ah very good point. I like this suggestion a lot, those names are a lot simpler and what you said about them being appended to the component name totally makes sense. I'll keep this in mind for the future!

File: airbyte-webapp/src/components/base/DropDown/CustomSelect.tsx [hunk 17]
```diff
@@ -36,8 +37,9 @@ export const CustomSelect = styled(Select)<
     }
 
     & .react-select__value-container {
-      overflow: visible;
+      overflow: hidden;
       display: flex;
+      flex-wrap: nowrap;
```
[the remark this answers is not in the document]
Comment (lmossman) [reply airbyte-webapp/src/components/base/DropDown/CustomSelect.tsx:42]: Without this change, the value in the dropdown control gets pushed up when the dropdown shrinks to a small enough size:

https://user-images.githubusercontent.com/22731524/190790238-584386f3-f7b0-4685-b7e2-44d713d8a53d.mov

With this change, the dropdown control value stays where it is even when it shrinks:

https://user-images.githubusercontent.com/22731524/190790827-09660805-1304-477a-827f-2fb202499c4d.mov

File: airbyte-webapp/src/components/GroupControls/GroupControls.module.scss [hunk 18]
```diff
@@ -0,0 +1,46 @@
+@use "../../scss/colors";
+
+$titleHeight: 34px;
+$groupSpacing: 15px;
```
[the remark this answers is not in the document]
Comment (lmossman) [reply airbyte-webapp/src/components/GroupControls/GroupControls.module.scss]: Fixed! Thanks

File: airbyte-webapp/src/components/base/Tooltip/InfoTooltip.module.scss [hunk 19]
```diff
@@ -4,16 +4,17 @@
...
   color: colors.$grey;
-  width: $icon-size;
+  width: calc($icon-size + $icon-padding);
+  padding-left: $icon-padding;
 }
 
 .icon {
-  position: absolute;
+  position: relative;
   top: $icon-top;
```
[the remark this answers is not in the document]
Comment (lmossman) [reply airbyte-webapp/src/components/base/Tooltip/InfoTooltip.module.scss:19]: Yeah I made sure to verify that the other places using this component looked correct, and they seem to look right to me.

I made these styling changes in this component to correct the info icon positioning in Safari and Firefox, since KC pointed out above that the positioning was off in those browsers.

For example, this is what the streams table in the replication view looks like in Safari when the `.icon` selector has `position: absolute`:
Screen Shot 2022-09-16 at 2 45 00 PM

With `position: relative`, the info icon positioning looks the same in Safari as it does in Firefox and Chrome:
Screen Shot 2022-09-16 at 2 46 04 PM

File: airbyte-webapp/src/views/Connector/ServiceForm/components/Sections/GroupLabel.module.scss [hunk 20]
```diff
@@ -0,0 +1,11 @@
+@use "../../../../../scss/colors";
+
+.groupLabel {
+  width: auto;
+  height: 100%;
+  padding-right: 10px;
```
[the remark this answers is not in the document]
Comment (lmossman) [reply airbyte-webapp/src/views/Connector/ServiceForm/components/Sections/GroupLabel.module.scss]: Good point! I've replaced that and a few other places with shared scss variables 👍

File: airbyte-webapp/src/views/Connector/ServiceForm/components/Sections/PropertyError.tsx [hunk 21]
```diff
@@ -0,0 +1,9 @@
+import React from "react";
```
[the remark this answers is not in the document]
Comment (lmossman) [reply airbyte-webapp/src/views/Connector/ServiceForm/components/Sections/PropertyError.tsx]: Yeah I think it does make more sense there, done 👍

File: airbyte-webapp/src/components/LabeledControl/ControlLabels.tsx [hunk 22]
```diff
@@ -29,9 +30,19 @@ const ControlLabels: React.FC<React.PropsWithChildren<ControlLabelsProps>> = (pr
...
+      {props.infoMessage && (
+        <InfoTooltip className={styles.tooltip} placement="top-start">
+          {props.infoMessage}
+        </InfoTooltip>
+      )}
+      {props.optional && (
+        <Text size="sm" className={styles.optionalText}>
+          <FormattedMessage id="form.optional" />
+        </Text>
+      )}
```
[the remark this answers is not in the document]
Comment (lmossman) [reply airbyte-webapp/src/components/LabeledControl/ControlLabels.tsx:42]: To me, the `<Label>` component felt like a more generic component that I wanted to keep pretty simple - basically it has the "content" (which are just the children you pass in), a "message", which is what gets displayed after the label content, and a couple other customization props. And this Label is used in this simple way in a lot of places.

The `ControlLabels` component to me felt like something that adds on top of the simple label: it offers the ability to add an info tooltip and an `Optional` string into the label "content". These two things felt like a more specific requirement for Form field labels specifically, which is where the `ControlLabel` component tends to get used, so I added those here. The `Optional` string specifically felt like it didn't make a ton of sense to add to `Label` since I think there are places that Label is used where optional strings don't really make sense, like Settings page.

However, I don't feel too strongly about this so I'm happy to consolidate all of this into the `Label` component if you think we should!

File: airbyte-webapp/src/components/LabeledControl/ControlLabels.tsx [hunk 23]
```diff
@@ -1,26 +1,27 @@
...
 export interface ControlLabelsProps {
   className?: string;
   error?: boolean;
   success?: boolean;
   nextLine?: boolean;
   message?: React.ReactNode;
-  errorMessage?: React.ReactNode;
   labelAdditionLength?: number;
   label?: React.ReactNode;
+  infoMessage?: React.ReactNode;
```
[the remark this answers is not in the document]
Comment (lmossman) [reply airbyte-webapp/src/components/LabeledControl/ControlLabels.tsx:19]: `infoMessage` was meant to contain specifically the content that should go in an InfoTooltip that is added to the label, if this prop is set. `message`, on the other hand, is the element that is displayed inline after the label itself. This PR moves a lot of the content that was in `message` into `infoMessage`, since that is part of the redesign, but other places in the app still use that `message` field.

Based on your comment, I've renamed this prop to `infoTooltipContent` which hopefully makes that a bit more clear.

File: airbyte-webapp/src/components/LabeledControl/ControlLabels.module.scss (hunk 15 above)
[the remark this answers is not in the document]
Comment (lmossman) [reply airbyte-webapp/src/components/LabeledControl/ControlLabels.module.scss]: Actually scratch that -- this does matter for labeled switch fields. If this is set to `block`, they look like this:
Screen Shot 2022-09-16 at 3 51 01 PM

And setting this to `inline-block` correctly centers the label:
Screen Shot 2022-09-16 at 3 53 33 PM

File: airbyte-webapp/src/views/Connector/ServiceForm/components/Property/PropertyLabel.tsx [hunk 24]
```diff
@@ -0,0 +1,42 @@
...
+
+interface PropertyLabelProps {
+  property: FormBlock;
+  label: React.ReactNode;
+  description?: string;
+  optional?: boolean;
+  className?: string;
+}
+
+const PropertyLabel: React.FC<React.PropsWithChildren<PropertyLabelProps>> = ({
```
Comment (krishnaglick) [inline airbyte-webapp/src/views/Connector/ServiceForm/components/Property/PropertyLabel.tsx:17]: Should export here, not below.

**Review by krishnaglick — COMMENTED** [review]
Really great work here Lake!
